src/JWT.Extensions.AspNetCore/JwtAuthenticationOptions.cs

```diff
@@ -77,5 +87,15 @@ public class JwtAuthenticationOptions : AuthenticationSchemeOptions
         /// The default value is <see cref="Dictionary{String, String}" />.
         /// </remarks>
         public Type PayloadType { get; set; } = typeof(Dictionary<string, object>);
+
+
+        /// <summary>
+        /// Custom Event Overrides
+        /// </summary>
+        public new JwtAuthenticationEvents Events

```

Comment:
Why `new` is needed?

Reply:
So the base class has the same property but it returns type `object` as seen here: https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/Core/src/AuthenticationSchemeOptions.cs#L33

You can see that Microsoft does the same thing with their OAuth https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/OAuth/src/OAuthOptions.cs#L70

So, I'm mostly copying this pattern. This allows you to add custom events that work just for you.

Reply:
Fair! I'm fine with doing this as long as this is a common pattern